cloudflared/DOCS.md

```diff
@@ -108,6 +108,10 @@ additional_hosts:
     service: "http://192.168.1.3:8080"
     disableChunkedEncoding: true
 nginx_proxy_manager: true
+strict_ssl: true
+generate_cert: true
+certfile: fullchain.pem
```

Comment:
Dont't think we need to make the name editable. we can simply both of them name "fullchain_cloudflare.pem" and "privkey_cloudflare.pem" or something like that, no no overrides are to be expected

Reply:
Hm... I disagree.

Every HTTPS guide in Home Assistant suggests using `/ssl/fullchain.pem` and `/ssl/privkey.pem`.

That's where the Let's Encrypt addon exports certificates to, and the DuckDNS addon as well.

Also, many addons comes configured by default to use the certificates placed in these locations as well. Changing them to something else would need users to manually tweak the (good) defaults of the existing addons, like Nginx proxy, TasmoAdmin, or any other HASS Community addon which serves HTTPS.

Also, it's unlikely for people to have two different addons generating certificates for them, like having both Let's Encrypt and Cloudflared.

But, if the user desires so for whatever reason, this is why allowing to change the default value would be good.

Reply:
Mhm fair enough, I am still not sure about the config option. Maybe another option would be so set the certificates to the standard-names as suggested by you (the point you make about the default config of other add-ons does make a lot of sense), but do not override and fail if there are already files with that name in the folder?

Reply:
> do not override and fail if there are already files with that name in the folder?

Perhaps the whole `generate_cert` should be disabled by default, and if users intentionally enable it, they will read in the documentation that it will overwrite existing files if they exist already.

By the way, overwriting already happens in the Let's Encrypt addon for a good reason: it automatically renews the certificate if the expiring date is approaching.

Reply:
Since HTTPS is not actually needed for enabling Strict SSL in Cloudflare (as you know, I initially thought the opposite), I think it's reasonable that we should not bother people about asking them to enable HTTPS in their HA, unless they want to. Thus, `generate_cert` would be disabled by default.

Reply:
Mhm not sure about this one. So they have to change something in the config anyways (so HA accepts the packages from Cloudflare). That means, that they just have to copy more into their config, which is not more work. At the same time, we would make it the standard case, that they have SSL also in their local network, which I feel is something that quite some people don't like about the current standard approach in this Add-on. So why not just make it standard to have SSL?